tests/test_cli_compile.py

```diff
@@ -1,42 +1,62 @@
 import os
 import subprocess
 import sys
+from functools import wraps
 from textwrap import dedent
 
 import mock
 import pytest
-from click.testing import CliRunner
 from pytest import mark
 
 from .constants import MINIMAL_WHEELS_PATH, PACKAGES_PATH
 from .utils import invoke
 
 from piptools._compat.pip_compat import PIP_VERSION, path_to_url
+from piptools.cache import DependencyCache
 from piptools.scripts.compile import cli
 
 
-def test_default_pip_conf_read(pip_with_index_conf, runner):
+@pytest.fixture
+def invoke_cli(runner, depcache_dir):
```

Comment:
instead of `invoke_cli` we can use the following fixture:
```python
@pytest.fixture(autouse=True)
def temp_dep_cache(tmpdir, monkeypatch):
    monkeypatch.setenv('PIP_TOOLS_CACHE_DIR', str(tmpdir / 'cache'))
```

so we don't have to touch all the below. lines. What do you think?